src/screen.js

```diff
@@ -0,0 +1,57 @@
+
+import config from './config'
+
+const AlpineScreenMagicMethod = {
+    start() {
+        Alpine.addMagicProperty('screen', function ($el) {
+            // bind resize event to window with debounce
+            let update
+            const updateScreen = () => {
+                clearTimeout(update)
+                update = setTimeout(() => {
+                    $el.__x.updateElements($el)
+                }, 150)
+            }
+
+            window.addEventListener('resize', updateScreen)
```

Comment:
There is a memory leak here. The helpers adds a new listener everytime screen gets evaluated and each listener triggers it's own component refresh so after a few interactions, there are loads of event listener attached to windows. This will be likely to make pages slow on a long run. you can add a console log inside the settimeout and see that the amount of log doubles for each alpine refresh

Reply:
Yes!! That's is what i found as well. Function call multiplied on each resize event. I even tried our good friend `once` method. I thought it's a known behavior so i forget to mention it. What would be the best path here?

Reply:
You'll need to keep track of it somehow so you don't attach multiple listeners for the same component. It will kill performance quickly on big pages.

Reply:
Not sure if it's proper way but i handled it by assigning a global variable on first init and check that to re-assign resize event.
```js
            let update
            const updateScreen = () => {
                clearTimeout(update)
                update = setTimeout(() => {
                    $el.__x.updateElements($el)
                }, 150)

                // set $screenHelperInitialized to prevent multiple calls
                window.$screenHelperInitialized = true
            }

            // bind resize event to window if not initialized
            if (!window.$screenHelperInitialized) {
                window.addEventListener('resize', updateScreen)
            }
```

Reply:
I'm not 100% sure (I haven't tried yet) but I feel like it won't work if you have 2 Alpine components using the helper.
For example
```html
<div x-data>
    <span x-show="$screen('lg')">Foo</span>
</div>
<div x-data>
    <span x-show="$screen('lg')">Bar</span>
</div>
```
I suspect only the first one will update.

Reply:
I might be wrong though 😂
Can you just double check pls?

Reply:
Yeah, i think i need to dive in deeper level for this part. But i think i got an idea. Let me try today or tomorrow.